### ISSUE TITLE
Default compiler does not work on windows (if ocide's compiler not found and correct system environment variables are set)

### ISSUE THREAD

**Issue by ColinDuquesnoy** [HackEdit/hackedit-cobol#12]
_From @brazitec on February 3, 2016 0:54_

select secondary compiler is not working
compiler is at g:\gnucobol but it goes to d:\gnucobol
Compiler configuration: 'Default'
G:\gnucobol\bin\cobc.EXE -x -o build\main.exe -std=cobol2002 -debug main.cbl
Configuration Error
D:\gnucobol\config;D:\gnucobol\config\cobol2002.conf: No such file or directory
cobc: Error: Invalid option -std=cobol2002
environment variables are set to g:\gnucobol

## System information

- Operating System: Windows-10-10.0.10586
- HackEdit: 1.0a2.dev7
- Python: 3.4.4 (64bits)
- Qt: 5.5.1
- PyQt: 5.5.1

_Copied from original issue: HackEdit/hackedit#54_

**Comment by ColinDuquesnoy** [comment]
Thank you for reporting!

But I am not sure this is a bug: "Default" configuration is currently broken on Windows. This is normal because that configuration is meant to use the compiler that is going to be bundled by the windows installer. Since there is no installer yet there is no bundled compiler. HE will try to use the compiler bundled with OpenCobolIDE if installed in its standard path. The fact that it picks up `d:\gnucobol` is probably because you set some system environment variables to point at `d:\gnucobol` and you don't have ocide installed in its standard path. **Can you confirm my guesses?**

"Default" configuration is meant to be read-only but I just noted that you can still edit the environment variables but their configuration is not saved/used which may lead to confusion... That is clearly a bug I want to fix.

I am not sure how this is related to "select secondary compiler" though. "Select secondary compiler" is used to compile a project/file with multiple compiler configurations. It will first compile with your primary compiler (which seems to be "Default" in your case) and then with the selected secondary compilers. You might want to add two compiler configuration derived from "Default" with correct settings. Choose one as your primary compiler and the other as the secondary compiler (i.e. don't use "Default" at all).

If I misunderstood the problem, please post a screenshot of the "Default" compiler configuration and the new compiler configuration you've set up and attach the application output log (`%APPDATA%\HackEdit\hackedit.log`)

**Comment by ColinDuquesnoy** [comment]
_From @brazitec on February 4, 2016 0:32_

Hello

  First thank you for responding to my error report.

I finally got the paths to match. 

    I am using gnucobol 2.0. I need to convert a micro focus system that uses the $set command that is not supported in 1.1

.

The only gnucobol directory is in the D:\gnucobol.

I still get the following error message

Compiler configuration: 'Default'

D:\gnucobol\bin\cobc.exe -x -o build\main.exe -std=default -debug main.cbl

Configuration Error

D:\gnucobol\config;D:\gnucobol\config\default.conf: No such file or directory

cobc: Error: Invalid option -std=default

 

these directories exist.

Hope you can help

**Comment by ColinDuquesnoy** [comment]
_From @brazitec on February 4, 2016 0:50_

here are the screen shots
![screenshot 1](https://cloud.githubusercontent.com/assets/6499340/12801961/f09f46b4-cac7-11e5-856c-094aff58aa4e.png)
![screenshot 2](https://cloud.githubusercontent.com/assets/6499340/12801960/f09bcb06-cac7-11e5-87c4-1e35b40c4a16.png)
![screenshot 3](https://cloud.githubusercontent.com/assets/6499340/12801962/f0a65530-cac7-11e5-8b05-46caf7c0aaf7.png)
[hackedit.txt](https://github.com/HackEdit/hackedit/files/116690/hackedit.txt)

**Comment by ColinDuquesnoy** [comment]
_From @brazitec on February 4, 2016 1:28_

log with debug info
[hackedit.txt](https://github.com/HackEdit/hackedit/files/116740/hackedit.txt)

**Comment by ColinDuquesnoy** [comment]
Looking at the log, I think I found the offender: 

``COB_CONFIG_DIR=D:\\gnucobol\\config;D:\\gnucobol\\config``

which leads to:

``D:\gnucobol\config;D:\gnucobol\config\default.conf: No such file or directory``

I can reproduce this from the command line with GnuCOBOL r658 on GNU/Linux:

```
[colin@nuc5i5ryh ~]$ export COB_CONFIG_DIR=/usr/share/gnu-cobol/config:/usr/share/gnu-cobol/config
[colin@nuc5i5ryh ~]$ cd dev/OpenCobolIDE/test/testfiles/
[colin@nuc5i5ryh testfiles]$ cobc -fsyntax-only HelloWorld.cbl
Configuration Error
/usr/share/gnu-cobol/config:/usr/share/gnu-cobol/config/default.conf: No such file or directory
```

The path to the config directory is defined twice and it seems like cobc does not like it. Did you set it that way? If no then there is a bug in HackEdit otherwise just set ``COB_CONFIG_DIR=D:\\gnucobol\\config`` and the compilation error should go away...

@GitMensch Any thoughts on this? Looks like a bug in cobc...

**Comment by ColinDuquesnoy** [comment]
_From @brazitec on February 5, 2016 0:17_

hello
  looks like its a bug in HE. I only have it defined once.
If you need any other screen shots let me know. The project i am converting is a large program for the accounting for a chain homeless shelters in Brazil. it will give HE a true work out....
looking forward to helping you get HE working on windows...
Gator

**Comment by ColinDuquesnoy** [comment]
I've managed to reproduce the issue. Should be fixed. Please upgrade using the same instruction I gave you in #52 (you should have version >= 1.0a2.dev15). Comment if you still have an issue and I'll reopen...

**Comment by ColinDuquesnoy** [comment]
_From @GitMensch on February 5, 2016 21:24_

From Colin:
> @GitMensch Any thoughts on this? Looks like a bug in cobc...

No bug, this is a "one directory only" setting. But I consider adding a nice error if a path separator is in :-)

From brazitec
> The project i am converting is a large program for the accounting for a chain homeless shelters in Brazil. it will give HE a true work out....

@brazitec 
This sounds interesting. If you haven't talked about this project at https://sourceforge.net/p/open-cobol/discussion I like to invite you to do so. Please check if it's possible to set this software under GPL or similar licenses, in this case I'd like to see it published in "Contributions" (BTW: there is already ACAS which is another accounting system).
The biggest issue in HE for big projects like this could be the indexer - does it work fine for you with the complete project being known to HE?

**Comment by GitMensch** [comment]
Just for back-reference: I've created a [feature request][1] ticket for checking DIR-entries for PATH separators at SF.

@brazitec: I'd still like to hear more of your project at https://sourceforge.net/p/open-cobol/discussion.

[1]:https://sourceforge.net/p/open-cobol/feature-requests/138/

**Comment by ColinDuquesnoy** [comment]
> Just for back-reference: I've created a feature request ticket for checking DIR-entries for PATH separators at SF.

Thank you? Will this be in 2.0 or post 2.0?

**Comment by GitMensch** [comment]
> > Just for back-reference: I've created a feature request ticket for checking DIR-entries for PATH separators at SF.

> Thank you! Will this be in 2.0 or post 2.0?

Was finished today and just merged to 2.0 :-)
It currently leads to an error (=abort) both for compiler and runtime configuration issues.

... and I've just recognized that my changed "exceptional" error messages (where the compiler itself aborts) doesn't match your error string checking.

Example (all these start with the prefix "cobc" and include the last line [here the second line] and return a status code of 99):

~~~~
cobc: environment variable 'COB_CONFIG_DIR' is '/temp:/tmp'; should not contain ':'
cobc: aborting
~~~~

**Comment by ColinDuquesnoy** [comment]
>  and I've just recognized that my changed "exceptional" error messages (where the compiler itself aborts) doesn't match your error string checking.
Example (all these start with the prefix "cobc" and include the last line [here the second line] and return a status code of 99):

No problem, I'll make the needed adaptations (#30 and OpenCobolIDE/OpenCobolIDE#335)
